moveit_core/robot_trajectory/test/robot_trajectory_test.cpp

```diff
@@ -0,0 +1,319 @@
+        "    <geometry>"
+        "      <box size=\"1 2 1\" />"
+        "    </geometry>"
+        "  </visual>"
+        "</link>"
+        "  <joint name=\"mim_f\" type=\"prismatic\">"
+        "    <axis xyz=\"1 0 0\"/>"
+        "    <limit effort=\"100.0\" lower=\"0.0\" upper=\"0.19\" velocity=\"0.2\"/>"
+        "    <parent link=\"link_c\"/>"
+        "    <child link=\"link_d\"/>"
+        "    <origin rpy=\" 0.0 0.0 0.0 \" xyz=\"0.1 0.1 0 \"/>"
+        "    <mimic joint=\"joint_f\" multiplier=\"1.5\" offset=\"0.1\"/>"
+        "  </joint>"
+        "  <joint name=\"joint_f\" type=\"prismatic\">"
+        "    <axis xyz=\"1 0 0\"/>"
+        "    <limit effort=\"100.0\" lower=\"0.0\" upper=\"0.19\" velocity=\"0.2\"/>"
+        "    <parent link=\"link_d\"/>"
+        "    <child link=\"link_e\"/>"
+        "    <origin rpy=\" 0.0 0.0 0.0 \" xyz=\"0.1 0.1 0 \"/>"
+        "  </joint>"
+        "<link name=\"link_d\">"
+        "  <collision>"
+        "    <origin rpy=\"0 0 0\" xyz=\"0 0 0\"/>"
+        "    <geometry>"
+        "      <box size=\"1 2 1\" />"
+        "    </geometry>"
+        "  </collision>"
+        "  <visual>"
+        "    <origin rpy=\"0 1 0\" xyz=\"0 0.1 0\"/>"
+        "    <geometry>"
+        "      <box size=\"1 2 1\" />"
+        "    </geometry>"
+        "  </visual>"
+        "</link>"
+        "<link name=\"link_e\">"
+        "  <collision>"
+        "    <origin rpy=\"0 0 0\" xyz=\"0 0 0\"/>"
+        "    <geometry>"
+        "      <box size=\"1 2 1\" />"
+        "    </geometry>"
+        "  </collision>"
+        "  <visual>"
+        "    <origin rpy=\"0 1 0\" xyz=\"0 0.1 0\"/>"
+        "    <geometry>"
+        "      <box size=\"1 2 1\" />"
+        "    </geometry>"
+        "  </visual>"
+        "</link>"
+        "</robot>";
+
+    static const std::string SMODEL2 =
+        "<?xml version=\"1.0\" ?>"
+        "<robot name=\"one_robot\">"
+        "<virtual_joint name=\"base_joint\" child_link=\"base_link\" parent_frame=\"odom_combined\" type=\"planar\"/>"
+        "<group name=\"base_from_joints\">"
+        "<joint name=\"base_joint\"/>"
+        "<joint name=\"joint_a\"/>"
+        "<joint name=\"joint_c\"/>"
+        "</group>"
+        "<group name=\"mim_joints\">"
+        "<joint name=\"joint_f\"/>"
+        "<joint name=\"mim_f\"/>"
+        "</group>"
+        "<group name=\"base_with_subgroups\">"
+        "<group name=\"base_from_base_to_tip\"/>"
+        "<joint name=\"joint_c\"/>"
+        "</group>"
+        "<group name=\"base_from_base_to_tip\">"
+        "<chain base_link=\"base_link\" tip_link=\"link_b\"/>"
+        "<joint name=\"base_joint\"/>"
+        "</group>"
+        "<group name=\"base_from_base_to_e\">"
+        "<chain base_link=\"base_link\" tip_link=\"link_e\"/>"
+        "<joint name=\"base_joint\"/>"
+        "</group>"
+        "<group name=\"base_with_bad_subgroups\">"
+        "<group name=\"error\"/>"
+        "</group>"
+        "</robot>";
```

Comment:
Is this URDF taken from another moveit file? Alternatively, you could use one of the URDFs from moveit_resources and load it from the file, to keep this test file concise.

Reply:
Yes -- it's taken from https://github.com/ros-planning/moveit/blob/melodic-devel/moveit_core/robot_state/test/robot_state_test.cpp

I'll look into the moveit_resources URDFs